components/org.wso2.carbon.identity.application.authz.xacml/src/main/java/org/wso2/carbon/identity/application/authz/xacml/handler/impl/XACMLBasedAuthorizationHandler.java

```diff
@@ -61,6 +61,9 @@
     public static final String ACTION_AUTHENTICATE = "authenticate";
     private static volatile XACMLBasedAuthorizationHandler instance;
 
+    private XACMLBasedAuthorizationHandler(){
```

Comment:
What is the usage of this constructor?

Reply:
The constructor was make as private because when using singleton pattern for the class, we should make constructor private. Otherwise objects can be created, without invoking getInstance method.